tardis/analysis_tools/line_ID.py

```diff
@@ -0,0 +1,274 @@
+"""
+This module provides a line identification tool that allows the user to extract
+a summary of the most prominent transitions in a TARDIS simulation.
+"""
+
+import numpy as np
+import pandas as pd
+import astropy.units as u
+import matplotlib.pyplot as plt
+
+from tardis.util.base import (
+    atomic_number2element_symbol,
+    int_to_roman,
+)
+
+# import the SDECData class from the sdec_plot module
+from tardis.visualization.tools.sdec_plot import SDECData
+
+
+class line_identifier:
+    def __init__(self, data):
+        """
+        Initialize line_identifier with required data of simulation model.
+
+        Parameters
+        ----------
+        data : dict of SDECData
+            Dictionary to store data required for SDEC plot, for both packet
+            modes i.e. real and virtual
+        """
+        self.data = data
+
+    @classmethod
+    def from_simulation(cls, sim):
+        """
+        Create an instance of line_identifier from a TARDIS simulation object.
+
+        Parameters
+        ----------
+        sim : tardis.simulation.Simulation
+            TARDIS Simulation object produced by running a simulation
+
+        Returns
+        -------
+        line_identifier
+        """
+        if sim.runner.virt_logging:
+            return cls(SDECData.from_simulation(sim, "virtual"))
+        else:
+            return cls(SDECData.from_simulation(sim, "real"))
+
+    def _reset_cache(self):
+        self._reset_lam_min()
+        self._reset_lam_max()
+        self._reset_derived_quantities()
+
+    def _reset_lam_min(self):
+        self._lam_min = None
+
+    def _reset_lam_max(self):
+        self._lam_max = None
+
+    def _reset_derived_quantities(self):
+        self._line_mask = None
+        self._lam_in = None
+        self._lines_info_unique = None
+        self._lines_count = None
+        self._lines_ids = None
+        self._lines_ids_unique = None
+
+    @property
```

Comment:
Can you please add docstring to this non-private property (an others below it)  so that docstring coverage pipeline passes?